src/client/options/mod.rs

```diff
@@ -547,6 +548,12 @@ pub struct ClientOptions {
     #[builder(default)]
     pub write_concern: Option<WriteConcern>,
 
+    /// The token callbacks for OIDC authentication.
+    #[builder(default)]
+    #[serde(skip)]
+    #[derivative(Debug = "ignore", PartialEq = "ignore")]
+    pub oidc_callbacks: Option<oidc::Callbacks>,
```

Comment:
Unless there are complications I'm not thinking of, I think it would be a little more consistent with other authentication methods to store these on the credential rather than the client options directly. WDYT?

As an aside, it would be nice to refactor our credential to be an enum with each mechanism as a variant that stores the information that's needed so that we can make more static guarantees about the absence/presence of fields, but that is a 3.0 concern 🙂